docs/source/hyper-parameter-search.rst

```diff
@@ -3,31 +3,82 @@ Hyper Parameter Search
 
 *Tools for performing hyperparameter optimization of Scikit-Learn API-compatible models using Dask*.
 
-There are two kinds of hyperparameter optimization estimators
-in Dask-ML. The appropriate one to use depends on the size of your dataset and
-whether the underlying estimator implements the `partial_fit` method.
+Issues in hyper-parameter searches
+----------------------------------
+Two scenarios can occur during hyper-parameter optimization. The
+hyper-parameter search can be
 
-If your dataset is relatively small or the underlying estimator doesn't implement
-``partial_fit``, you can use :class:`dask_ml.model_selection.GridSearchCV` or
-:class:`dask_ml.model_selection.RandomizedSearchCV`.
-These are drop-in replacements for their scikit-learn counterparts, that should offer better performance and handling of Dask Arrays and DataFrames.
-The underlying estimator will need to be able to train on each cross-validation split of the data.
-See :ref:`hyperparameter.drop-in` for more.
+1. compute constrained
+2. memory constrained
 
-If your data is large and the underlying estimator implements ``partial_fit``, you can
-Dask-ML's :ref:`*incremental* hyperparameter optimizers <hyperparameter.incremental>`.
+These issues are independent and both can happen the same time. Being memory
+constrained has to do with dataset size, and being compute constrained has to
+do with estimator complexity and number of possible hyper-parameter
+combinations.
+
+An example of being compute constrained is with almost any neural network or
+deep learning framework. An example of being memory constrained is when the
+dataset doesn't fit in RAM. Dask-ML covers all 4 combinations of these two
+constraints.
+
+Neither compute nor memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Scikit-learn handles this case.
```

Comment:
Though, note that Dask-ML's pipeline re-writing / caching can still be helpful here.